node/src/actors/chain_manager/handlers.rs

```diff
@@ -596,6 +598,15 @@ impl PeersBeacons {
         // Some(None) (most of the peers did not send a beacon)
         .and_then(|x| *x)
     }
+
+    /// Collects the peers to unregister based on the beacon they reported and the beacon to be compared it with
+    pub fn decide_peers_to_unregister(&self, beacon: CheckpointBeacon) -> Vec<SocketAddr> {
+        // Unregister peers which have a different beacon
+        (&self.pb)
+            .iter()
+            .filter_map(|(p, b)| if *b != Some(beacon) { Some(*p) } else { None })
```

Comment:
This is believed to be more efficient, as it short-circuits as soon as `p == None`:
```suggestion
            .filter_map(|(peer, b)| *b.filter(|b| b != beacon).map(|_| *peer))
```

Reply:
thanks!

Reply:
Be careful, this is not equivalent. If `b` is `None`, it should return `Some(*p)` because we want to unregister peers that have sent no beacon.